lib/syslog/writer.go

```diff
@@ -244,6 +247,23 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
+func cachedDialWriter(network string, address string, config *tls.Config, socksProxy string) (w io.Writer, err error) {
+	// We don't currently include the TLS configuration in the cache key. For
+	// now, this is probably OK as the TLS configuration is unlikely to change.
+	// If we really want to be correct, we can use
+	// github.com/mitchellh/hashstructure to get a hash of the config for
```

Comment:
Actually this wouldn't work because `tls.Config` contains functions, which this package doesn't know how to hash.